server/src/main/java/org/elasticsearch/search/lookup/LeafDocLookup.java

```diff
@@ -75,7 +83,12 @@ public void setDocument(int docId) {
             scriptValues = AccessController.doPrivileged(new PrivilegedAction<ScriptDocValues<?>>() {
                 @Override
                 public ScriptDocValues<?> run() {
-                    return fieldDataLookup.apply(fieldType).load(reader).getScriptValues();
+                    // TODO should this go through QueryShardContext?
+                    IndexFieldData<?> ifd = fieldDataLookup.apply(fieldType);
+                    if (ifd instanceof SearchLookupAware) {
+                        ((SearchLookupAware) ifd).setSearchLookup(searchLookup);
+                    }
+                    return ifd.load(reader).getScriptValues();
```

Comment:
Yeah, that is what my TODO was about - maybe this should go through `getForField`. I can look into making that change in master. I think it is good regardless. And, if it has consequences that we didn't think of we may as well see them earlier rather than later...

Reply:
I don't know if this makes sense on master and what difference it would make, but I think it does make sense in our feature branch to isolate the hack around augmenting the fielddata impl.